platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblemsSummary.kt

```diff
@@ -119,17 +127,21 @@ class ConfigurationCacheProblemsSummary(
                 ProblemSeverity.Interrupting -> {}
             }
             if (overflowed) {
+                // we already overflowed during a previous problem
                 return false
             }
-            if (totalProblemCount > maxCollectedProblems) {
+            if (problemCauses.size == maxCollectedProblems) {
+                // we are overflowing now
                 overflowed = true
                 return false
             }
-            if (severity != ProblemSeverity.SuppressedSilently) {
```

Comment:
Before, we would not apply deduplication to problems with this severity, now we do

Reply:
🤔 Do we cover this by integTest?